convert2rhel/pkghandler.py

```diff
@@ -1144,52 +1111,165 @@ def clean_yum_metadata():
+
+    :return: Return a number indicating if the versions match, are less or greater then.
     :rtype: int
     """
+    # call parse_pkg_string to obtain name, epoch, release, version and arch fields
+    # order of fields: name, epoch, version, release, arch
+    version1_components = parse_pkg_string(version1)
+    version2_components = parse_pkg_string(version2)
+
+    # ensure package names match, error if not
+    if version1_components[0] != version2_components[0]:
+        raise ValueError("%s does not match %s" % (version1_components[0], version2_components[0]))
+
+    # ensure package arches match, error if not
+    if version1_components[4] != version2_components[4] and all(([version1_components[4]], version2_components[4])):
+        raise ValueError("%s does not match %s" % (version1_components[4], version2_components[4]))
+
+    # create list containing EVR for comparison
+    evr1 = (version1_components[1], version1_components[2], version1_components[3])
+    evr2 = (version2_components[1], version2_components[2], version2_components[3])
+
+    return rpm.labelCompare(evr1, evr2)
+
+
+def parse_pkg_string(pkg):
+    """
+    This function takes a version string in NEVRA, NEVR, NVRA, NVR, ENVRA, ENVR and decides whether
+    to parse with a yum/dnf module based on the package manager type of the system
 
-    # TODO(r0x0d): This function still needs some enhancements code-wise, it
-    # workes perfectly, but the way we are handling the versions is not 100%
-    # complete yet.  will be done in a future work. Right now, all the list of
-    # changes are listed in this comment:
-    # https://github.com/oamg/convert2rhel/pull/469#discussion_r873971400
-    pkg_ver_components = []
-    for pkg in pkg_1, pkg_2:
-        # Remove the package name and split the rest between epoch + version
-        # and release + arch
-        epoch_version, release_arch = pkg.rsplit("-", 2)[-2:]
-        # Separate the (optional) epoch from the version
-        epoch_version = epoch_version.split(":", 1)
-        if len(epoch_version) == 1:
-            epoch = "0"
-            version = epoch_version[0]
+    :return: Return a Return a list containing name, epoch, version, release, arch
+    :rtype: list[str | None]
+    """
+    if pkgmanager.TYPE == "yum":
+        pkg_ver_components = _parse_pkg_with_yum(pkg)
+    else:
+        pkg_ver_components = _parse_pkg_with_dnf(pkg)
+
+    return pkg_ver_components
+
+
+def _parse_pkg_with_yum(pkg):
+    """Parse verison string using yum and rpmUtils splitFilename
+
+    :return: Return a list containing name, epoch, version, release, arch
+    :rtype: list[str | None]
+    """
+
+    # package is in NEVRA/NEVR format
+    if re.findall(r"-\d:", pkg):
+        name, epoch_version, release_arch = pkg.rsplit("-", 2)
+        epoch, version = epoch_version.split(":", 1)
+        # package is in NEVRA format
+        if re.findall(r".*(x86_64|s390x|i86|ppc64le|aarch64)$", release_arch):
```

Comment:
release_arch contains both release and arch, on the next line the string gets split into the release and arch fields

Reply:
I think using regex is a bit better too since it detects that the arch comes directly after a "."

Reply:
This could be done with groups in RegEx too. Though I'd vote for not having RegEx here and instead go with something like this. Since we're looking at the end of the line
```suggestion
        if release_arch.endswith((x86_64, s390x, i86, ppc64le, aarch64)):
```

Reply:
Thanks @SpyTec I'll add this!